src/vs/editor/contrib/suggest/suggestWidget.ts

```diff
@@ -695,6 +691,11 @@ export class SuggestWidget implements IContentWidget, IListVirtualDelegate<Compl
 				this.focusedItem = null;
 				break;
 			case State.Open:
+				if (!this.editor.getConfiguration().contribInfo.suggest.showIcons) {
+					addClass(this.element, 'no-icons');
+				} else {
+					removeClass(this.element, 'no-icons');
+				}
```

Comment:
It's okay to add/remove class here in `setState` if `State.Open`? I couldn't find another solution and I had to move it out from `constructor` because after setting change reload is required, with this change you're able to change the setting without reload. Let me know if it's OK.

Reply:
I would leave the code in the ctor and listen to a config-change event from the editor, e.g. `editor.onDidChangeConfiguration`. Whenever the config changes that event fires and then you can update the state.

Reply:
Good suggestion!

1. I should do the same for `maxSuggestionsToShow`?
2. I should do it like `configureFont()` in `Renderer.renderTemplate()` with `Event.chain<IConfigurationChangedEvent>(this.editor.onDidChangeConfiguration.bind(this.editor))`?

Reply:
I think that it's OK for `maxSuggestionsToShow`. The `conifgureFont` is a little extra complicated for my taste 😄. I would, in `SuggetWidget` itself, add a "normal" listener, like

```ts
editor.onDidChangeConfiguration((e => {
	if(e.contribInfo) {
		// update icon-class
	}
}))
```

Reply:
Oh, and don't forget to add the listener-handle to the `this.toDispose`

Reply:
Okay, I'm doing this right now. Thank you very much for help!

Reply:
Quick question, but still I have to access `this.editor.getConfiguration().contribInfo.suggest` in `onDidChangeConfiguration`?

```js
editor.onDidChangeConfiguration((e => {
	if(e.contribInfo) {
		const showIcons = this.editor.getConfiguration().contribInfo.suggest.showIcons;
	}
}))
```

Reply:
Yeah, that should work

Reply:
@jrieken Already pushed new commit. Let me know if a new implementation is correct.